lib/model/model.go

```diff
@@ -1243,20 +1243,69 @@ func (m *Model) updateLocals(folder string, fs []protocol.FileInfo) {
 		return
 	}
 	files.Update(protocol.LocalDeviceID, fs)
-
+	
 	filenames := make([]string, len(fs))
 	for i, file := range fs {
 		filenames[i] = file.Name
 	}
-
+	
+	// Lets us know if file/folder change was originated locally or from a network
+	// sync update.  Now write these to a global log file.
+	if remoteUpdate != true {
+		m.writeToGlobalLog(m.folderCfgs[folder].Path(), fs)
+	}
+	
 	events.Default.Log(events.LocalIndexUpdated, map[string]interface{}{
-		"folder":    folder,
-		"items":     len(fs),
-		"filenames": filenames,
-		"version":   files.LocalVersion(protocol.LocalDeviceID),
+		"folder":       folder,
+		"items":        len(fs),
+		"filenames":    filenames,
+		"version":      files.LocalVersion(protocol.LocalDeviceID),
+		"remoteupdate": remoteUpdate,	
 	})
 }
 
+func (m *Model) writeToGlobalLog(path string, files []protocol.FileInfo) {
+	now := time.Now()
+	path = strings.Replace(path, "\\\\?\\", "", 1)
+	separator := "/"
+	if runtime.GOOS == "windows" {
+		separator = "\\"
+	}
+	
+	// Strip off the last forward/backslash (and filename) from ConfigPath and append our log file instead
+	logPath := m.cfg.ConfigPath()[:strings.LastIndex(m.cfg.ConfigPath(), separator)] + separator + "filemods.log"
+	
+	// Now open the file for writing (create without error if not existing)
+	f, err := os.OpenFile(logPath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+	
+	objType := " file"
+	action := " modified"
```

Comment:
These should be in the loop, otherwise if you process a deleted directory, and then in the next loop you happen to hit a file modified, it would still be reported as deleted directory, just because you don't reset the values back.